nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/model/state/internal/DefaultParamState.java

```diff
@@ -732,27 +732,30 @@ public void setVisible(boolean visible) {
 		
 		if (null == findIfNested().getParams()) {
 			return;
-		}
+		}  
 		
-		traverseChildren(p -> p.setVisible(visible));
+		if(this.getConfig().getUiStyles()==null) {
```

Comment:
This will make the entity state different between a UI based and non UI based param. Should we make changes to not traverse the child and handle in the UI that if parent is set to not visible then hide all child components

Reply:
This is added to handle params -> which are UI based but don't have a ViewConfig added. For eg : In the following example - we don't need to traverse the children for SampleSection and SampleForm, but would need it for SampleNonUIInnerClass, since server does'nt send the config for this param to UI. We wont be able to handle this through UI traversing because the server doesnt send the parent config for SampleNonUIInnerClass for UI to interpret.

@Section
private SampleSection sampleSection ;

public class SampleSection{
    private SampleNonUIInnerClass sampleNonUIInnerClass;
}

public class SampleNonUIInnerClass{

   @Form
   private SampleForm sampleForm;
}